include/h2o.h

```diff
@@ -1962,7 +1962,14 @@ inline h2o_conn_t *h2o_create_connection(size_t sz, h2o_context_t *ctx, h2o_host
     conn->ctx = ctx;
     conn->hosts = hosts;
     conn->connected_at = connected_at;
+#ifdef H2O_NO_64BIT_ATOMICS
+    static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+    pthread_mutex_lock(&mutex);
+    conn->id = ++h2o_connection_id;
+    pthread_mutex_unlock(&mutex);
+#else
```

Comment:
I am afraid that with this code `mutex` would be created for every source file that includes h2o.h since the function is declared as `static`.

IIUC we need to define the mutex as a global variable in one of our source files (presumably lib/common/memory.c) and use it.

Reply:
I guess you're right. I'll fix this, thanks!